static/app/components/events/searchBar.tsx

```diff
@@ -110,23 +111,75 @@ function SearchBar(props: SearchBarProps) {
               item =>
                 !Object.keys(FIELD_TAGS).includes(item.field) && !isEquation(item.field)
             )
-            .map(item => [item.field, {key: item.field, name: item.field}])
+            .map(item => [
+              item.field,
+              {key: item.field, name: item.field, kind: FieldValueKind.FUNCTION},
+            ])
         )
       : {};
 
-    const fieldTags = organization.features.includes('performance-view')
-      ? Object.assign({}, measurements, FIELD_TAGS, functionTags)
-      : omit(FIELD_TAGS, TRACING_FIELDS);
-
-    const combined = assign({}, tags, fieldTags, SEMVER_TAGS);
-    combined.has = {
+    const fieldTags = Object.fromEntries(
+      Object.keys(FIELD_TAGS).map(key => [
+        key,
+        {
+          ...FIELD_TAGS[key],
+          kind: FieldValueKind.FIELD,
+        },
+      ])
+    );
+
+    const measurementsWithKind = Object.fromEntries(
+      Object.keys(measurements).map(key => [
+        key,
+        {
+          ...measurements[key],
+          kind: FieldValueKind.MEASUREMENT,
+        },
+      ])
+    );
+
+    const combinedTags: Record<string, Tag> = organization.features.includes(
+      'performance-view'
+    )
+      ? Object.assign({}, measurementsWithKind, fieldTags, functionTags)
+      : omit(fieldTags, TRACING_FIELDS);
+
+    const semverTags = Object.fromEntries(
+      Object.keys(SEMVER_TAGS).map(key => [
+        key,
+        {
+          ...SEMVER_TAGS[key],
+          kind: FieldValueKind.FIELD,
+        },
+      ])
+    );
+
+    const tagsWithKind = Object.fromEntries(
+      Object.keys(tags).map(key => [
+        key,
+        {
+          ...tags[key],
+          kind: FieldValueKind.TAG,
+        },
+      ])
+    );
+
+    assign(combinedTags, tagsWithKind, fieldTags, semverTags);
```

Comment:
We now have to add `kind` to every key so we can display the type on the right side of the searchbar. However, this won't be needed in a near future PR where we consolidate all the field definitions along with defining its `kind`.